src/sentry/sentry_metrics/querying/metadata/tags.py

```diff
@@ -0,0 +1,60 @@
+from collections.abc import Sequence
+from dataclasses import dataclass
+
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+from sentry.sentry_metrics.use_case_id_registry import UseCaseID
+from sentry.snuba.metrics_layer.query import fetch_metric_tag_keys, fetch_metric_tag_values
+
+
+@dataclass
+class TagValue:
+    key: str
+    value: str
+
+    def __hash__(self):
+        return hash((self.key, self.value))
+
+
+def get_tag_keys(
```

Comment:
Are you using this function in this PR?

Reply:
This is an initial implementation of the function get_tag_keys as a basis for a future PR that will then use this to replace the get_all_tags from snuba/metrics/datasource.py - that logic has proven very hard to detangle, therefore I'd like to move it to a PR of its own.